docs/server-shutdown.rst

```diff
@@ -0,0 +1,30 @@
+======================
+ Server Shutdown Flow
+======================
+
+This document details the sequence of operations executed when Rafka server receives a shutdown
+signal. The steps executed by the `Server` are summarized in the following list:
+
+#. Receives the shutdown request
+#. Stops accepting new `Consumers`
+#. Closes all `Clients` with an active `Consumer` instance
+#. Waits for all `Consumers` to terminate
+#. Stops accepting new `Clients`
```

Comment:
I think we should rephrase this to `Stops accepting new `Producers`, for clarity.

Reply:
It actually stops accepting new Clients not Producers in this step. You can't initialize a connection with Rafka after this step, so changing it to _Stop accepting new Producers_ would confuse a user because it would except its connection be accepted.

I'd say to leave this as is for now.